tuf/api/metadata.py

```diff
@@ -859,22 +882,69 @@ def to_dict(self) -> Dict[str, Any]:
         }
 
 
+class TargetFile:
+    """A container with information about a particular target file.
+
+    Attributes:
+        length: An integer indicating the length of the target file.
+        hashes: A dictionary mapping hash algorithms to the
+            hashes resulting from applying them over the metadata file
+            contents::
+
+              'hashes': {
+                    '<HASH ALGO 1>': '<TARGET FILE HASH 1>',
+                    '<HASH ALGO 2>': '<TARGET FILE HASH 2>',
+                    ...
+                }
+        unrecognized_fields: Dictionary of all unrecognized fields.
+
+    """
+
+    @property
+    def custom(self):
+        if self.unrecognized_fields is None:
+            return None
+        return self.unrecognized_fields.get("custom", None)
+
+    def __init__(
+        self,
+        length: int,
+        hashes: Dict[str, str],
+        unrecognized_fields: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        self.length = length
+        self.hashes = hashes
+        self.unrecognized_fields = unrecognized_fields
+
+    @classmethod
+    def from_dict(cls, target_dict: Dict[str, Any]) -> "TargetFile":
+        """Creates TargetFile object from its dict representation."""
+        length = target_dict.pop("length")
+        hashes = target_dict.pop("hashes")
+        # All fields left in the target_dict are unrecognized.
+        return cls(length, hashes, target_dict)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Returns the JSON-serializable dictionary representation of self."""
+        res_dict = {
+            "length": self.length,
+            "hashes": self.hashes,
+        }
+
+        if self.custom is not None:
+            res_dict["custom"] = self.custom
```

Comment:
Should you have unrecognised_fields handling here instead of handling custom?

Reply:
I realized that I don't need to handle custom specifically any more.
"custom", if exists, will be part of `unrecognized fields and my `to_dict` function could be just:
```
    def to_dict(self) -> Dict[str, Any]:
        """Returns the JSON-serializable dictionary representation of self."""
        return {
            "length": self.length,
            "hashes": self.hashes,
            **self.unrecognized_fields,
        }
```